src/Utils.php

```diff
@@ -288,7 +300,7 @@ public static function jsonDecode(string $json, bool $assoc = false, int $depth
     public static function jsonEncode($value, int $options = 0, int $depth = 512): string
     {
         $json = \json_encode($value, $options, $depth);
-        if (JSON_ERROR_NONE !== \json_last_error()) {
+        if (JSON_ERROR_NONE !== \json_last_error() || false === $json) {
```

Comment:
IMHO opinion this is a problem of PHPStan

https://github.com/phpstan/phpstan/issues/3317

If I already checked a function that can tell me if `$json`  is false.. why do I need to check `$json` as well..

Reply:
Yep.

Reply:
Any idea on how to prevent the PHPStan error from happening? Would it help to write unit tests to cover that `false === $json` assertion?

Reply:
Do you have a test case that the `\json_last_error()` will return `JSON_ERROR_NONE`  but the json would still be false?

if yes pls post on 3v4l.org/ first...

Reply:
Agreed, as encoding `false` will result in `string(5) "false"`

reverted/fixed in 6b3389102af334bec698edeff60513a6dd142444